### PR TITLE
Add Twitter as OAuth provider

### PR DESCRIPTION
**Pull request by pbrisbin**
- Extract fromProfileURL and reuse

### CODE REVIEW

**Comment by pbrisbin** [comment]
Ping @thoughtbot/haskell

File: Yesod/Auth/OAuth2.hs [hunk 1]
```diff
@@ -104,5 +108,21 @@ authOAuth2 name oauth getCreds = AuthPlugin name dispatch login
         <a href=@{tm $ oauth2Url name}>Login via #{name}
         |]
 
+-- | Handle the common case of fetching Profile information a JSON endpoint
+--
+-- Throws @'InvalidProfileResponse'@ if JSON parsing fails
+--
+handleProfileResponse :: FromJSON a
```
Comment (jferris) [inline Yesod/Auth/OAuth2.hs]: Is there a better word we can use here than "handle?" Map? Decode?

Comment (pbrisbin) [inline Yesod/Auth/OAuth2.hs]: I wouldn't say it does either of those things. It's already decoded when we get the `a` and `map` is very overloaded. Maybe "convert"? I chose "handle" for "error handling".

Comment (jferris) [inline Yesod/Auth/OAuth2.hs]: Maybe `tryAuth' or `authOrFail`?

Comment (pbrisbin) [inline Yesod/Auth/OAuth2.hs]: I'm not used to worrying about naming so much in Haskell. Can I just call it `f`?

How about `fromProfileURL`?

Comment (jferris) [inline Yesod/Auth/OAuth2.hs]: Haha, `f`.

Yeah, I think `fromProfileURL` is pretty good.

**Comment by pbrisbin** [comment]
Ready to merge with that rename?

**Comment by jferris** [comment]
Looks good!

**Pull request merged**